### PR TITLE
THREESCALE-2297: Adding PF4-React login components

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:
Adding Patterfly 4 - React Login Page

https://issues.jboss.org/browse/THREESCALE-2297

**Verification steps** 

- Pull this branch
- Remove node_modules `rm -rf node_modules`
- Install packages `npm install`
- Start server and go to login page 

**Screenshots:**

**Large screen:**
login-fullscreen

**Small screen:**
login-small-screen

**With Authentication providers:**
login-sso

**Enforcing SSO:**
login-enforce-sso

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,12 +42,13 @@
     "eslint-plugin-standard": "^4.0.0",
     "file-loader": "^1.1.6",
     "flow-bin": "^0.93.0",
-    "flow-typed": "^2.5.1",
+    "flow-typed": "=2.5.1",
```
Comment (damianpm) [inline package.json]: This is to fix the `Error: Malformed npm package name!` error, look at [here](https://github.com/flow-typed/flow-typed/issues/3363)

File: spec/javascripts/LoginPage/LoginPageWrapper.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  authenticationProviders: null,
+  providerAdminDashboardPath: 'provider-admin-path',
+  providerLoginPath: 'provider-login-path',
+  providerPasswordPath: 'password-path',
+  providerSessionsPath: 'sessions-path',
+  redirectUrl: 'redirect-url',
+  show3scaleLoginForm: true
+}
+
+it('should render itself with right props', () => {
```
Comment (didierofrivia) [inline spec/javascripts/LoginPage/LoginPageWrapper.spec.jsx]: I'd rather test where those props are used or how they modify the component. I don't see the value in testing the the previously defined object that was passed to the component and asserting its values.

File: spec/javascripts/LoginPage/LoginPageWrapper.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  expect(wrapper.props().authenticationProviders).toEqual(null)
+  expect(wrapper.props().providerAdminDashboardPath).toEqual('provider-admin-path')
+  expect(wrapper.props().providerLoginPath).toEqual('provider-login-path')
+  expect(wrapper.props().providerPasswordPath).toEqual('password-path')
+  expect(wrapper.props().providerSessionsPath).toEqual('sessions-path')
+  expect(wrapper.props().redirectUrl).toEqual('redirect-url')
+  expect(wrapper.props().show3scaleLoginForm).toEqual(true)
+})
+
+it('should call getURL method on mount', () => {
```
Comment (didierofrivia) [inline spec/javascripts/LoginPage/LoginPageWrapper.spec.jsx]: if we changed this method and adopt also DI for the window object, you could assert the functionality of changing the state of the form with different href values

File: app/views/provider/sessions/new.html.slim [hunk 4]
```diff
@@ -1,32 +1,12 @@
...
-          li.forgot
-            = link_to 'Forgot password?', provider_login_path(request_password_reset: true), class: 'toggle-panel', data: { panel:  'request_password_reset_panel' }
-
-          li.commit
-            = submit_tag 'Sign in', class: 'important-button'
-
-    = render 'request_password_reset_form'
+= javascript_pack_tag 'LoginPage'
+- authentication_providers = (@authentication_providers || []).map { |ap| {authorize_url: ap.authorize_url, human_kind: ap.human_kind} }
+- flash_messages = (flash || []).map { |f| {type: f[0], message: f[1]}}
```
Comment (damianpm) [inline app/views/provider/sessions/new.html.slim]: @Martouta please check this :)

File: .flowconfig [hunk 5]
```diff
@@ -13,4 +13,4 @@
 [options]
 module.name_mapper.extension='scss' -> 'empty/object'
 module.system=haste
-module.name_mapper='\(Applications\|Dashboard\|Navigation\|Onboarding\|Policies\|services\|Stats\|Types\|Users\|utilities\|NewService\)\(.*\)$' -> '<PROJECT_ROOT>/app/javascript/src/\1/\2'
+module.name_mapper='\(Applications\|Dashboard\|LoginPage\|Navigation\|Onboarding\|Policies\|services\|Stats\|Types\|Users\|utilities\|NewService\)\(.*\)$' -> '<PROJECT_ROOT>/app/javascript/src/\1/\2'
```
[the remark this answers is not in the document]
Comment (damianpm) [reply .flowconfig]: It fails with `Unable to parse line` if you add break lines

File: app/views/provider/sessions/new.html.slim (hunk 4 above)
Comment (Martouta) [reply app/views/provider/sessions/new.html.slim]: sorry but no time for me to do reviews during this week 😄

File: app/views/layouts/provider/login.html.slim [hunk 6]
```diff
@@ -5,9 +5,9 @@ html lang="en"
     title 3scale Login
     = csrf_meta_tag
 
-    = render 'provider/theme'
     = render 'provider/analytics'
-
+    = stylesheet_link_tag 'loginPageFonts'
+    = javascript_pack_tag 'pf4BaseStyles'
```
[the remark this answers is not in the document]
Comment (damianpm) [reply app/views/layouts/provider/login.html.slim]: For some reason, if I try to use `stylesheet_pack_tag` webpacker is unable to add it to the manifest.json file. I'll keep it as a js import by now and do further research

File: app/javascript/packs/pf4BaseStyles.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+/*  Import all PF4 styles */
+// import '@patternfly/react-core/dist/styles/base.css'
```
[the remark this answers is not in the document]
Comment (damianpm) [reply app/javascript/packs/pf4BaseStyles.js]: Look comment [here](https://github.com/3scale/porta/pull/761/files#diff-329d6a7669b5c4857e240c9f0690b22fR10)

**Comment by didierofrivia** [comment]
There are some easy to fix codeclimate issues there, including frozen strings in ruby files, some repetition in js, double quotes and other minor language feats that we might want to tackle before merging. After that, please, squash the commits ;) 
Good work! 🎖

**Pull request merged**